### ISSUE TITLE
Apparent bug using colspan to form column groups

### ISSUE THREAD

**Issue by Leprosy** [wenzhixin/bootstrap-table#2051]
Hi guys, excellent library...however, I'm having an issue making column groups with `colspan` attribute

Here is a fiddle, http://jsfiddle.net/tdow66ef/1/

If you see the code, the second column of `group a`, named `a-two`, is hidden. But the first column of group b is taking it's place. If you toggle `a-two`'s visibility, it returns to it's normal, desired position.

**Comment by djhvscf** [comment]
This issue is fixed in https://github.com/wenzhixin/bootstrap-table/pull/2006 by @dabros

**Comment by Leprosy** [comment]
Is this code in the last version of bootstrap-table? I'm using 1.10.1 in the fiddle.

**Comment by djhvscf** [comment]
No, this code is from #2006 but it is still opened.. It will be released in some days..

See this example: http://jsfiddle.net/djhvscf/tdow66ef/3/

**Comment by Leprosy** [comment]
Great, thanks!

**Comment by Leprosy** [comment]
Looks great except for this case: at the http://jsfiddle.net/djhvscf/tdow66ef/3/ turn on all columns and deactivate the last two ones(`b-three`, `b-fou`r, or both)

**Comment by djhvscf** [comment]
@Leprosy what exactly is the error there?

**Comment by Leprosy** [comment]
Like I said, this happens if you turn off any of the two last columns(I did it after turning on all columns of the tables). Here is a screenshot(Chrome Version 48.0.2564.116 (64-bit) under Ubuntu 14.04):

![table](https://cloud.githubusercontent.com/assets/851905/13528496/1563271a-e1f5-11e5-98f0-b5819f17ceda.png)

If you see the image, the header of the last group has now a `colspan=2`, when it must be 3

**Comment by Leprosy** [comment]
You can check out this fiddle too: I have 3 levels of headers, but the first one has some weird span(should be colspan=6)

http://jsfiddle.net/w18w4t1e/1/

**Comment by dabros** [comment]
Re. linked pull request: Fix #1578 : group-columns not updating on data-visible #2006

Getting caught up at work so cant finish atm, but am looking for opinion on underlying issue (as there is other code breaking my pull code, but not my inline fiddle - traced it down there)

Hopefully get some thoughts from others before i can finish, since i want to normalise all calls to similar code to avoid other known and potential issues too.

**Comment by Leprosy** [comment]
I patched up the code and now it works correctly(both issues I've fiddled now are ironed out).

Replacing line 736 of `bootstrap-table.js` with this:

```javascript
        $.each(this.options.columns, function(i, columns) {
            $.each(columns, function(j, column) {
                column.colspan = 0;
            })
        });

        $.each(this.options.columns[this.options.columns.length - 1], function(i, column) {
            column.colspan = 1;

            if (typeof column.parents !== 'undefined') {
                $.each(column.parents,function(k,lookupStr) {
                    var lookupKeys = lookupStr.split("."),
                        parentCol = that.options.columns[lookupKeys[0]][lookupKeys[1]];
                        parentCol.colspan += (column.visible) ? 1 : 0; //(parentCol.colspan > 1) ? -1 : 0;
                        parentCol.visible = (parentCol.colspan > 0);
                });
            }
        });
```
